lib/ValidateDirectiveVisitor.ts

```diff
@@ -219,8 +219,17 @@ const checkMustValidateInput = (
   if (finalType instanceof GraphQLInputObjectType) {
     const fieldsToCheck = Object.values(finalType.getFields()).filter(field => {
       if (field.type instanceof GraphQLList) {
+        if (field.type.ofType instanceof GraphQLNonNull) {
+          return finalType !== field.type.ofType.ofType;
+        }
+
         return finalType !== field.type.ofType;
       }
+
+      if (field.type instanceof GraphQLNonNull) {
+        return field.type.ofType !== finalType;
```

Comment:
yeah, this will cause problems with `[T!]!`, we need to go recursive here as well, see lines above.

Reply:
done, `[T!]!` will be handled properly